File: web-development/level-1/helsinki-part-1-2/Python/part1/week04/ex83/HangmanLogic.py
class HangmanLogic:

    #consturctor
    def __init__(self, word):
        self.word = word
        self.guessed_letters = []
        self.number_of_faults = 0

    #behaviors

    #gets the nubmer of faults
    def get_number_of_faults(self):
        return self.number_of_faults

    #gets the guessed letters
    def get_guessed_letters(self):
        return self.guessed_letters

    #limit for the wrong guesses
    def losing_fault_amount(self):
        return 12

    #guessing a letter
    def guess_letter(self, letter):
        # add to the guesses
        self.guessed_letters += letter
        # if letter is already been guessed nothing happens
        if letter in self.guessed_letters:
            print('You have already guessed this letter: ', letter)
        #checking if the letter is wrong
        if letter not in self.word:
            self.number_of_faults += 1

    #hiding the word
    def hidden_word(self):
        for i in self.word:
            print('_')

File: web-development/level-1/helsinki-part-1-2/Java/part1/week4/week04-ex83-GameLogicOfHangman/HangmanLogic.java
public class HangmanLogic {

  //fields
  private String word;
  private String guessedLetters;
  private int numberOfFaults;

  //constructor
  public HangmanLogic(String word) {
    this.word = word.toUpperCase();
    this.guessedLetters = "";
    this.numberOfFaults = 0;
  }

  //behaviors
  public int numberOfFaults() {
    return this.numberOfFaults;
  }

  public String guessedLetters() {
    return this.guessedLetters;
  }

  public int losingFaultAmount()  {
    return 12;
  }

  public void guessLetter(String letter) {
       // program here the functionality for making a guess
       // if the letter has already been guessed, nothing happens
       if (this.word.contains(letter.toUpperCase()))  {
       }
       // it the word does not contains the guessed letter, the number of faults increases
       if (!this.word.contains(letter.toUpperCase())) {
         this.numberOfFaults++;
       }
       // the letter is added among the already guessed letters
       this.guessedLetters = this.guessedLetters + letter;
   }

}
